source/dev/idea_list.md

```diff
@@ -0,0 +1,148 @@
+- The tool ensures that the data is correct (format, data types, value domains, missing values, etc.), and either cleans the input data or responds with an error
+- The tool should be well documented with a focus on code quality
+
+**Requirement:** Python
+
+**Level:** Easy
+
+**Mentor:**
+
+### Project 3. - Simulation Data Ingestion Pipeline
+To run energy system simulations we need input data such as electrical demand, hydro flow, solar irradiance, and wind speed time series data among others. See [here](https://github.com/Breakthrough-Energy/PreREISE#2-gather-data-for-simulation) for more details . 
+Currently the data is collected by using the package when needed. There is still a lot of manual labor involved in collecting and cleaning the data. Besides that the data source or API’s may change.
+
+The goal of this project is to automate the download, validation, and archival of external resources using industry standard components such as Apache Airflow, InfluxDB, Grafana, Blob Stores, etc. 
+
+Note that this project will eventually incorporate the work of the Simulation Data Validation Manager and Simulation Data Download Manager projects. However, each of these projects can proceed in parallel.
+
+
+- Write Apache Airflow workflows that periodically 
+    - Query external data sources for availability
+    - Download external data sources
+    - Validate and clean the downloaded resources
+    - Archive downloaded resources to a blob store
+- Workflows are instrumented with telemetry
+- Telemetry are aggregated in InfluxDB
+- Metrics are displayed in a Grafana dashboard
+- If a data source is down, we throw an alarm
+The workflows should be well documented with a focus on code quality, and using best-of-breed current technology stacks (Apache Airflow, Kubernetes, Helm, Docker, Python3, Grafana, etc.)
+
+**Requirements:**
+Docker, Python, Grafana, PAAS, Apache Airflow
+
+**Level:** 
+Medium to Hard
+
+**Mentor:**
+
+### Project 4. - Test system for end-to-end testing of the simulation framework 
+Use [existing test power systems](http://labs.ece.uw.edu/pstca/) that will be used in our simulation framework. 
+Generate input data to feed to the simulation engine using data science techniques.
+Write integration tests and produce documentation to showcase our open source software.
+
+- Collect data (python scripting)
+- Generate time series for demand, hydro, solar and wind matching the test grid (python data analysis)
+- Integrate the test model in our [simulation framework](https://github.com/Breakthrough-Energy/PowerSimData) (python)
+- Write end-to-end tests (in docker container)
+- Write GitHub actions to test changes of the framework when pushed on GitHub
+- Write documentation for the community (sphinx)
+
+**Requirements:**
+Object Oriented Programming in Python, Data analysis/plotting in Python (pandas, numpy, matplotlib, etc.) and docker
+
+**Level:** 
+Easy
+
+**Mentor:**
+[Ben](https://github.com/rouille)
```

Comment:
Feel free to reference me by my github profile as well:

Todd https://github.com/toddg

Reply:
@ToddG which projects should I add you?
I assume you are the primary mentor for #4

Reply:
@kasparm can we have several names under one project? If yes, you can add @ToddG here.